src/main/java/com/example/wdmsystem/employee/system/EmployeeService.java

```diff
@@ -30,7 +30,7 @@ public Employee createEmployee(CreateEmployeeDTO request) {
         String hashedPassword = passwordEncoder.encode(request.password());
         Employee employee = new Employee(
                 0,
-                currentUser.getMerchantId(),
+                currentUser.getMerchant(),
```

Comment:
Here and in other cases the method should still be getMerchantId(), as its better to store one int in the security context than a whole entity. The way that id is assigned the first time is what will have to change.